py-polars/polars/lazyframe/frame.py

```diff
@@ -155,7 +155,11 @@ def _gpu_engine_callback(
     new_streaming: bool,
     _eager: bool,
 ) -> Callable[[Any, int | None], None] | None:
-    is_gpu = (is_config_obj := isinstance(engine, GPUEngine)) or engine == "gpu"
+    is_gpu = (
+        (is_config_obj := isinstance(engine, GPUEngine))
+        or engine == "gpu"
+        or get_engine_affinity() == "gpu"
+    )
```

Comment:
I think you don't need to make this change if changing the default `engine` argument to `None` below.

I think `get_engine_affinity()` should only be called if `engine is None`, but here `engine` cannot be `None`.